### PR TITLE
[JENKINS-10860] Log failures in isMavenVersionValid but do not block other activities

### PR DESCRIPTION
**Pull request by jglick**
See [JENKINS-10860](https://issues.jenkins-ci.org/browse/JENKINS-10860). Arguably does not count as a full fix, since it does not address the root causes of the exceptions, but at least prevents them from breaking everything.

### CODE REVIEW

**Comment by jglick** [comment]
@reviewbybees

File: src/main/java/org/jfrog/hudson/maven3/extractor/MavenExtractorEnvironment.java [hunk 1]
```diff
@@ -96,7 +96,12 @@ public void buildEnvVars(Map<String, String> env) {
         }
 
         // if not valid Maven version don't modify the environment
-        if (!isMavenVersionValid()) {
+        try {
+            if (!isMavenVersionValid()) {
+                return;
+            }
+        } catch (Exception e) {
+            e.printStackTrace(buildListener.error("Unable to determine Maven version"));
```
Comment (oleg-nenashev) [inline src/main/java/org/jfrog/hudson/maven3/extractor/MavenExtractorEnvironment.java]: It would be better to use a common <code>LOGGER</code> approach

**Comment by oleg-nenashev** [comment]
:bee: 
The issue seems to be a PR builder glitch

**Comment by oleg-nenashev** [comment]
Yes, the new build was OK.

File: src/main/java/org/jfrog/hudson/maven3/extractor/MavenExtractorEnvironment.java (hunk 1 above)
Comment (amuniz) [inline src/main/java/org/jfrog/hudson/maven3/extractor/MavenExtractorEnvironment.java]: Then the message would go to regular logs not the build log

**Comment by amuniz** [comment]
:bee:

**Comment by jglick** [comment]
@reviewbybees done

**Pull request merged**